audittools/backing_store_file.go

```diff
@@ -0,0 +1,460 @@
+// SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company
+// SPDX-License-Identifier: Apache-2.0
+
+package audittools
+
+import (
+	"bufio"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"slices"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/majewsky/gg/option"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/sapcc/go-api-declarations/cadf"
+
+	"github.com/sapcc/go-bits/logg"
+)
+
+// NewFileBackingStore creates a file-based backing store from JSON parameters.
+// This is the factory function for use in AuditorOpts.BackingStoreFactories.
+//
+// Example usage:
+//
+//	auditor, err := NewAuditor(ctx, AuditorOpts{
+//	    BackingStoreFactories: map[string]BackingStoreFactory{
+//	        "file": NewFileBackingStore,
+//	    },
+//	})
+func NewFileBackingStore(params json.RawMessage, opts AuditorOpts) (BackingStore, error) {
+	var store FileBackingStore
+	if err := json.Unmarshal(params, &store); err != nil {
+		return nil, fmt.Errorf("audittools: failed to parse file backing store config: %w", err)
+	}
+
+	registry := opts.Registry
+	if registry == nil {
+		registry = prometheus.DefaultRegisterer
+	}
+
+	if err := store.Init(registry); err != nil {
+		return nil, err
+	}
+	return &store, nil
+}
+
+// FileBackingStore implements BackingStore using local filesystem files.
+// Provides durable audit buffering for services with persistent volumes.
+//
+// Thread safety: Write() and ReadBatch() are serialized by a mutex.
+// Multiple concurrent calls are safe but will block each other.
+// Callers must ensure that commit() completes before the next ReadBatch() call.
+type FileBackingStore struct {
```

Comment:
Same as for InMemoryBackingStore: All the public methods on this type are the ones that implement BackingStore, so this type does not need to be public. Only `NewFileBackingStore` needs to be public.